libs/parallelism/algorithms/include/hpx/parallel/algorithms/transform_exclusive_scan.hpp

```diff
@@ -307,13 +488,104 @@ namespace hpx { namespace parallel { inline namespace v1 {
+      : hpx::functional::tag_fallback<transform_exclusive_scan_t>
+    {
+        // clang-format off
+        template <typename InIter, typename OutIter, typename T,
+            typename BinOp, typename UnOp,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_iterator<InIter>::value &&
+                hpx::traits::is_iterator<OutIter>::value &&
+                hpx::is_invocable_v<UnOp,
+                    typename std::iterator_traits<InIter>::value_type> &&
+                hpx::is_invocable_v<BinOp,
+                    typename hpx::util::invoke_result<UnOp,
+                        typename std::iterator_traits<InIter>::value_type>::type,
+                    typename hpx::util::invoke_result<UnOp,
+                        typename std::iterator_traits<InIter>::value_type>::type
+                >
+            )>
+        // clang-format on
+        friend OutIter tag_fallback_dispatch(hpx::transform_exclusive_scan_t,
+            InIter first, InIter last, OutIter dest, T init, BinOp&& binary_op,
+            UnOp&& unary_op)
+        {
+            static_assert((hpx::traits::is_input_iterator<InIter>::value),
+                "Requires at least input iterator.");
+            static_assert((hpx::traits::is_output_iterator<OutIter>::value),
+                "Requires at least output iterator.");
+
+            return hpx::parallel::v1::detail::transform_exclusive_scan<
+                OutIter>()
+                .call(hpx::execution::seq, first, last, dest,
+                    std::forward<UnOp>(unary_op), std::move(init),
+                    std::forward<BinOp>(binary_op));
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter1, typename FwdIter2,
+            typename T, typename BinOp, typename UnOp,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter1>::value &&
+                hpx::traits::is_forward_iterator<FwdIter2>::value &&
+                hpx::is_invocable_v<UnOp,
+                    typename std::iterator_traits<FwdIter1>::value_type> &&
+                hpx::is_invocable_v<BinOp,
+                    typename hpx::util::invoke_result<UnOp,
+                        typename std::iterator_traits<FwdIter1>::value_type>::type,
+                    typename hpx::util::invoke_result<UnOp,
+                        typename std::iterator_traits<FwdIter1>::value_type>::type
+                >
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy,
+            FwdIter2>::type
+        tag_fallback_dispatch(hpx::transform_exclusive_scan_t,
+            ExPolicy&& policy, FwdIter1 first, FwdIter1 last, FwdIter2 dest,
+            T init, BinOp&& binary_op, UnOp&& unary_op)
+        {
+            static_assert(hpx::traits::is_forward_iterator<FwdIter1>::value,
```

Comment:
If you have `is_forward_iterator` in the concept-requires clause above, this `static_assert` will be superfluous. We usually check for `is_iterator` in the concept-requires clause and for the actual iterator category required here to improve error messages.